File: src/main/java/com/javawarriors/buyerside/services/SellerQnAService.java
package com.javawarriors.buyerside.services;

import com.javawarriors.buyerside.entities.*;
import com.javawarriors.buyerside.repositories.*;

import org.hibernate.cache.spi.support.AbstractReadWriteAccess.Item;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.*;

/**
 * Service layer for handling logic related to buyer QnA
 */
@Service
public class SellerQnAService {
    @Autowired
    private SellerQnARepo sellerQnARepo;

    @Autowired
    private IFSService ifsService;

    public SellerQnA saveSellerQnA(SellerQnA sellerQnA) {
        return sellerQnARepo.save(sellerQnA);
    }

    public List<SellerQnA> findByIfsListing(Long ifsId) {
        ItemForSaleListing ifsListing = ifsService.findByListingId(ifsId);
        return sellerQnARepo.findByIfsListing(ifsListing);
    }

}

File: src/main/frontend/src/services/SellerQnAService.js
import axios from "axios";
import authHeader from "../util/authHeader";
import { API_BASE_URL } from "../util/apiBaseUrl";

const SELLER_QNA_API_BASE_URL = `${API_BASE_URL}/seller-qna`;

class SellerQnAService {
  postSellerQnA(sellerQnA) {
    return axios.post(SELLER_QNA_API_BASE_URL + "/post", sellerQnA, {
      headers: authHeader(),
    });
  }

  getQnAsForIfsListing(ifsId) {
    return axios.get(SELLER_QNA_API_BASE_URL + "/get/ifslisting=" + ifsId, {
      headers: authHeader(),
    });
  }
}

export default new SellerQnAService();
